### DISCUSSION TITLE
Opensnitch with different user accounts

### DISCUSSION THREAD

**Discussion by ifernandolopez** [evilsocket/opensnitch#719] [Q&A] [answered]
Hello,

The user interface is automatically launched in the application in /home/flopez/.config/autostart/opensnitch_ui.desktop

When you switch to another user account the UI of opensnitch does not launch properly with the following error message:

```
 /bin/sh -c "pkill -15 opensnitch-ui; opensnitch-ui"
E0807 13:47:57.924487811   10152 server_chttp2.cc:40]        {"created":"@1659872877.924447092","description":"No address added out of total 1 resolved","file":"src/core/ext/transport/chttp2/server/chttp2_server.cc","file_line":307,"referenced_errors":[{"created":"@1659872877.924440806","description":"Unable to configure socket","fd":32,"file":"src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":216,"referenced_errors":[{"created":"@1659872877.924434101","description":"OS Error","errno":98,"file":"src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":190,"os_error":"Address already in use","syscall":"bind"}]}]}
```

I think that the problem is with the ""Address already in use" message

¿Is it possible to use opensnitch from different user accounts or are there any plan to do it possible?

Thank you

**Comment by gustavo-iniguez-goya** [comment, marked as answer]
hi @ifernandolopez , it's not possible to use the GUI from a different user account **while** the GUI is running on another user session.

The GUI is composed of 2 components: a grpc server and the GUI itself. In order to use the Qt GUI from different user accounts (local or remote), we should:
- split the GUI in 2 components
- make the grpc server a system service
- connect to the grpc server when the systray icon is launched
- and implement some kind of user authentication.

Decoupling the server from the UI would allow us to have different UIs (TUI, GUI, web UI, ...). I worked on it a couple of years ago, but I haven't had time to update it https://github.com/gustavo-iniguez-goya/opensnitch/tree/cli/server

**Reply by ifernandolopez** [reply]
Thank you Gustavo,

I mean to use the GUI from a different user account **in the local machine**. I am saying this aiming to reduce the bullet of work described above.

**Comment by johnny-smitherson** [comment]
will necro-post answer here for others that are looking for it

you can add a script to each user to monitor for "switch user" actions using dbus monitor + systemd-login events on session going Active and Inactive. You can then use this to kill the gui on the inactive user and start it on the active user. Here is a short script

```bash
#!/bin/bash

set -e

# Get our specific Session ID
SESSION_ID=$(loginctl session-status | head -n 1 | awk '{print $1}')

# Track the last known state
LAST_STATE=$(loginctl show-session "$SESSION_ID" -p Active --value)

while true; do
    # Query current state
    CURRENT_STATE=$(loginctl show-session "$SESSION_ID" -p Active --value)

    if [[ "$CURRENT_STATE" != "$LAST_STATE" ]]; then
        if [[ "$CURRENT_STATE" == "yes" ]]; then
            # --- ENTERING SESSION ---
            # Run the script for when you come back from the login screen
            echo "SWITCH TO ACTIVE"
            nohup opensnitch-ui --background > /tmp/opensnitch-gui-$SESSION_ID.log &
        else
            # --- EXITING SESSION ---
            # Run the script for when you switch away to the login screen
            echo "SWITCH TO INACTIVE"
            killall opensnitch-ui || true
        fi
        LAST_STATE="$CURRENT_STATE"
        echo "STATE IS $LAST_STATE"
    fi

    # Poll every 1 second
    sleep 1
done


```

run this script on both users on their respective login (e.g. with systemd --user)
